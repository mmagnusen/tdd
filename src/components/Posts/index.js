import React, { Component } from 'react';

class Posts extends Component {

  render() {
      const { posts = [] } = this.props;

  return (
        <section className='Blog-posts'>
            <h4>Posts:</h4>
            {posts.length === 0 ? <p>No posts</p> : posts.map((post) => <h3 key={post.title.rendered}>{post.title.rendered}</h3>)}
        </section>
  );
}}

export default Posts;
