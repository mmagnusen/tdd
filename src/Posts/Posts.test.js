import React from 'react';
import renderer from 'react-test-renderer'
import data from '../__mocks__/data'
import Posts from './';

describe('Posts component', () => {

    describe('when rendered', () => {

        it('renders without crashing if there are no posts', () => {
            const posts = renderer.create(<Posts/>)
            expect(posts.toJSON()).toMatchSnapshot()
        })

        it('renders, showing posts if posts are passed as props', () => {
            const posts = renderer.create(<Posts posts={data}/>)
            expect(posts.toJSON()).toMatchSnapshot()
        })
    })
});
