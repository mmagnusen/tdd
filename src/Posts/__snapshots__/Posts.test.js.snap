// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component when rendered renders without crashing if there are no posts 1`] = `
<section
  className="Blog-posts"
>
  <h4>
    Posts:
  </h4>
  <p>
    No posts
  </p>
</section>
`;

exports[`Posts component when rendered renders, showing posts if posts are passed as props 1`] = `
<section
  className="Blog-posts"
>
  <h4>
    Posts:
  </h4>
  <h3>
    1. title with post created from axios
  </h3>
  <h3>
    2. title with post created from axios
  </h3>
  <h3>
    3. title with post created from axios
  </h3>
</section>
`;
